src/parser/Expression.ts

```diff
@@ -280,7 +281,7 @@ export class FunctionParameterExpression extends Expression {
         public equalsToken?: Token,
         public defaultValue?: Expression,
         public asToken?: Token,
-        public typeToken?: Token,
+        public typeExpr?: TypeExpression,
```

Comment:
Thinking we can call this `type`, remove the `type: BscType` parameter, and instead leverage the `getType()` metohd on `TypeExpression`?

Reply:
It's a little harder with this one, because the type of the parameter is derived either from the explicit type given (e.g. `as float`) OR from the initial value, and the initial value could be a functionCall, etc...